packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -1001,6 +1001,14 @@ export const Table = <
                       column.minWidth = columnHasExpanders
                         ? COLUMN_MIN_WIDTHS.withExpander
                         : COLUMN_MIN_WIDTHS.default;
+
+                      if (
+                        column.width &&
+                        typeof column.width === 'number' &&
+                        column.minWidth > column.width
+                      ) {
+                        column.minWidth = column.width;
```

Comment:
Maybe I'm misunderstanding something, but wasn't the problem deemed to be within our resizing logic? Then why does this PR not touch the resizing logic at all? Iirc you said something along the lines of: our resize logic updates the `style` but does not update the actual `column.width`.

Reply:
I think this is coming from a misunderstanding. The logic correctly updates both the `style` and the `column.width`, but the issue is that the logic breaks if the `width` is less than the `minWidth`, which causes the `column.width` to not update until the user resizes by enough to make up the difference. This is what causes the shift, since the actual width of the column itself doesn't change, but the resize logic still takes the difference from later columns on the table anyways. Since the issue is coming from the `column.width` value being less than the `column.minWidth`, it makes sense to prevent this from happening at initialization.

Reply:
It makes sense when you say "the resize logic still takes the difference from later columns on the table anyways", but can you point to where in code this is happening? I'm having trouble finding it.

Reply:
It starts with the [`dispatchMove()`](https://github.com/iTwin/iTwinUI/blob/714b874246b458231b5ceddc8acd551c022b4743/packages/itwinui-react/src/core/Table/hooks/useResizeColumns.tsx#L101) function. This takes in the amount that the [mouse has moved on the x-axis](https://github.com/iTwin/iTwinUI/blob/714b874246b458231b5ceddc8acd551c022b4743/packages/itwinui-react/src/core/Table/hooks/useResizeColumns.tsx#L111) and passes it to the column resizing action. In this issue, this initially does nothing, since the resizing action recognizes that the new width that the first column is trying to be given is invalid. However, once the first columns new width becomes valid, the action is still observing how much the mouse has [moved to decide how much the later columns need to have taken from their widths](https://github.com/iTwin/iTwinUI/blob/714b874246b458231b5ceddc8acd551c022b4743/packages/itwinui-react/src/core/Table/hooks/useResizeColumns.tsx#L295). This causes a shift, since the first column's width isn't actually being updated, it is being "increased" to the size that it already was.